### PR TITLE
Opaque meta module + test - colorbar

### PR DESCRIPTION
**Pull request by tech4GT**
(replaces #762) -- copying in description from there:

Re #315 -- an attempt to refactor the colorbar module to be a fully opaque meta module - that is, a module which:

1. contains other modules
2. acts like a normal module in every way
3. contains an independent sequencer in order to pass an input through a sequence and produce an output

If this works (and it's not running yet -- I'm stuck for some reason because `draw()` is not executing... help!) then we may be able to create a generator for these, which is to say, a way to automate the creation of opaque meta modules given:

1. a JSON object or string of steps
2. a function which maps the outer module options to the options objects of the internal steps (or falling back to the defaults if mapping is not provided)
3. a set of overriding defaults for internal steps (which could be done via the mapping function too)

@publiclab/is-reviewers take a look here, i got inspired and tried it out, but need a bit of help to debug my work!

### CODE REVIEW

**Comment by tech4GT** [comment]
@jywarren Actually there is no way of testing the options.inBrowser thing since we run the tests in node and inBrowser is false there anyway, haha.
Anyway I have fixed the condition and I don't think we need a test here, I tried all cases manually though. :)

**Comment by tech4GT** [comment]
Long term:
@jywarren I just had an Idea!! If we think scale we can bootstrap this `sequencer inside sequencer` idea to multiple image use cases like map-knitter, where we can initialize one instance of Sequencer for each image and run all of them as independent processes!!
Wouldn't this parallelize our workflow?? Eager to hear your thoughts!! :smiley:

**Comment by jywarren** [comment]
(re:#762)

Re parallelize, EXACTLY!!!

Re test, can we just do a looks_like comparison with a visually confirmed output image? I just think any test for colorbar would be good. Let's confirm this can be added as a step, it produces an output, and that output looks like a visually confirmed output image. 

Re circular, I saw your comment and we can try that after this, yeah. Great!

**Comment by jywarren** [comment]
Also, what went wrong with this Travis build? Thanks!!

**Comment by tech4GT** [comment]
@jywarren Writing up the test and docs for this now!
Let's make the generator in another PR, also please do take a look at the optimizations pr!
Thanks!

File: src/modules/Colorbar/Module.js [hunk 1]
```diff
@@ -8,21 +8,19 @@ module.exports = function Colorbar(options, UI) {
...
     { 'name': 'gradient', 'options': {} },
     { 'name': 'colormap', 'options': { colormap: options.colormap } },
     { 'name': 'crop', 'options': { 'y': 0, 'h': options.h } },
     { 'name': 'overlay', 'options': { 'x': options.x, 'y': options.y, 'offset': -4 } }
   ];
 
-  var internalSequencer = ImageSequencer({ inBrowser: false });
-  console.log('colorbar setup is', internalSequencer);
+  // ui: false prevents internal logs
+  var internalSequencer = ImageSequencer({ inBrowser: false, ui: false });
```
Comment (jywarren) [inline src/modules/Colorbar/Module.js]: We should document this too!

**Comment by jywarren** [comment]
this is great, thanks @tech4GT !!!

**Comment by jywarren** [comment]
OK - now see: #765!

**Pull request merged**